test/ibmq/test_jupyter.py

```diff
@@ -81,12 +81,13 @@ def _limit_jobs(**kwargs):
             kwargs['skip'] = 5
             kwargs['start_datetime'] = datetime.now() - timedelta(days=7)
             return original_backend_jobs(**kwargs)
-
         for backend in self.backends:
             with self.subTest(backend=backend):
-                original_backend_jobs = backend.jobs
-                backend.jobs = _limit_jobs
-                jobs_tab(backend)
+                provider = backend.provider()
+                provider_backend_jobs = provider.backend.jobs(backend_name=backend.name())
+                original_backend_jobs = provider_backend_jobs
+                provider_backend_jobs = _limit_jobs
+                jobs_tab(provider.backend)
```

Comment:
This isn't working like before. Please test again.

Reply:
How should this be refactored? We want to set the jobs method from `IBMQBackendService` instead of the one from `IBMQBackend` to `_limit_jobs` but still have to filter by the backend name somehow if we're looping through the backends.  We also want to pass an `IBMQBackend` as a parameter to `jobs_tab` to keep things consistent - so I'm not sure how that would work. Simply doing something like
```
provider = backend.provider()
original_backend_jobs = provider.backend.jobs
```
would result in a recursion error

Reply:
We have to use `IBMQBackend` as a parameter in `jobs_tab` and then change the `_job_summary` code to 
```
provider = backend.provider()
jobs = provider.backend.jobs(backend_name=backend.name(), limit=None, start_datetime=past_year_date)
```
and then in the test case `_limit_jobs` method should be the stub for `provider.backend.jobs`.

Reply:
I haven't tested this but something like this should work? Could you try it?

```python
def test_jobs_tab(self):
        """Test jobs tab."""
        def _limit_jobs(**kwargs):
            kwargs['limit'] = 5
            kwargs['skip'] = 5
            kwargs['start_datetime'] = datetime.now() - timedelta(days=7)
            return original_backend_jobs(**kwargs)

        for backend in self.backends:
            with self.subTest(backend=backend):
                provider = backend.provider()
                original_backend_jobs = provider.backend.jobs
                provider.backend.jobs = _limit_jobs
                jobs_tab(backend)
```

Reply:
@kt474 Let's do this for now. Should be fine to change the signature since `jobs_tab` method is not used directly by our users.

## Code
```python
from typing import Any, Union
```
```python
def _job_summary(backend: Union[IBMQBackend, FakeBackend], **kwargs: Any) -> PlotlyWidget:
    """Interactive jobs summary for a backend.

    Args:
        backend: Display jobs summary for this backend.
        **kwargs: Used only for testing purposes:

            * limit - Number of jobs to retrieve.
            * start_datetime - Filter by the given start date, in local time. This is used to
                find jobs whose creation dates are after (greater than or equal to) this
                local date/time.

    Returns:
        A figure for the rendered job summary.
    """
    now = datetime.datetime.now()
    past_year_date = now - datetime.timedelta(days=365)
    limit = kwargs.pop('limit', None)
    start_datetime = kwargs.pop('start_datetime', past_year_date)
    provider = backend.provider()
    jobs = provider.backend.jobs(backend_name=backend.name(),
                                 limit=limit, start_datetime=start_datetime)
```
```python
def jobs_tab(backend: Union[IBMQBackend, FakeBackend], **kwargs: Any) -> wid.HBox:
    """Construct a widget containing job information for an input backend.

    Args:
        backend: Input backend.
        **kwargs: Used only for testing purposes:

            * limit - Number of jobs to retrieve.
            * start_datetime - Filter by the given start date, in local time. This is used to
                find jobs whose creation dates are after (greater than or equal to) this
                local date/time.

    Returns:
        An widget containing job summary.
    """
    title = wid.HTML('<h4>Click graph to display jobs</h4>')
    table = wid.HTML('', layout=wid.Layout(max_height='500px',
                                           height='500px',
                                           width='100%',
                                           overflow='hidden scroll',))

    sun_wid = _job_summary(backend, **kwargs)
```

## Test case
```python
def test_jobs_tab(self):
        """Test jobs tab."""
        for backend in self.backends:
            with self.subTest(backend=backend):
                limit = 5
                start_datetime = datetime.now() - timedelta(days=7)
                jobs_tab(backend, limit=limit, start_datetime=start_datetime)
```